packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -99,6 +98,147 @@ export function TelemetryView(): React.ReactElement {
 		};
 	}, [messageRelay, setTelemetryEvents]);
 
+	return (
+		<Stack>
+			<StackItem>
+				<ListLengthSelection
+					currentLimit={maxEventsToDisplay}
+					onChangeSelection={(key): void => setMaxEventsToDisplay(key)}
+				/>
+			</StackItem>
+			<StackItem>
+				{telemetryEvents !== undefined ? (
+					<FilteredTelemetryView telemetryEvents={telemetryEvents} />
+				) : (
+					<Waiting label={"Waiting for Telemetry events"} />
+				)}
+			</StackItem>
+		</Stack>
+	);
+}
+
+/**
+ * {@link ListLengthSelection} input props.
+ */
+interface ListLengthSelectionProps {
+	/**
+	 * The current limit (max number of telemetry events to show).
+	 * @defaultValue {@link DEFAULT_PAGE_SIZE}
+	 */
+	currentLimit: number;
+
+	/**
+	 * Called when the selection changes.
+	 */
+	onChangeSelection(newLimit: number): void;
+}
+
+/**
+ * A dropdown menu for selecting how many logs to display on the page.
+ */
+function ListLengthSelection(props: ListLengthSelectionProps): React.ReactElement {
+	const { currentLimit, onChangeSelection } = props;
+	const stackTokens: IStackTokens = { childrenGap: 20 };
+
+	// Options formatted for the Fluent Dropdown component
+	const dropdownOptions: { key: number; text: string }[] = [
+		{ key: 50, text: "50" },
+		{ key: 100, text: "100" },
+		{ key: 500, text: "500" },
+		{ key: 1000, text: "1000" },
+	];
+
+	const handleMaxEventChange: DropdownProps["onOptionSelect"] = (event, data) => {
+		onChangeSelection(Number(data.optionText));
+	};
+
+	return (
+		<Stack tokens={stackTokens}>
+			<div style={{ marginLeft: "6px" }}>
+				<h3>Max number of telemetry events to display: </h3>
+				<Dropdown
+					placeholder="Select an option"
+					size="small"
+					style={{ minWidth: "300px", zIndex: "1" }}
+					defaultValue={currentLimit.toString()}
+					// change the number of logs displayed on the page
+					onOptionSelect={handleMaxEventChange}
+				>
+					{dropdownOptions.map((option) => {
+						return (
+							<Option style={{ minWidth: "120px" }} key={option.key}>
+								{option.text}
+							</Option>
+						);
+					})}
+				</Dropdown>
+			</div>
+		</Stack>
+	);
+}
+
+/**
+ * {@link FilteredTelemetryView} input props.
+ */
+interface FilteredTelemetryViewProps {
+	/**
+	 * The current limit (max number of telemetry events to show).
+	 * @defaultValue {@link DEFAULT_PAGE_SIZE}
+	 */
+	telemetryEvents: ITimestampedTelemetryEvent[] | undefined;
```

Comment:
Does this need to allow `undefined`? Since the parent component handles the Waiting case, I think this can just be:
```suggestion
	telemetryEvents: ITimestampedTelemetryEvent[];
```